### DISCUSSION TITLE
useQuasar for Quasar v1/Vue 2 with composition api?

### DISCUSSION THREAD

**Discussion by lyle45** [quasarframework/quasar#8973] [General - Components / Directives / etc] [answered]
Hi, is there a composition hook like useQuasar for Quasar v1/Vue 2 with composition api, or is the only way to access quasar instance is like so?

``` typescript
  async setup(props, ctx) {    
    const q = ctx.root.$q    
  },
```

**Comment by metalsadman** [comment, marked as answer]
Nope, no useQuasar in qv1. You can import from quasar what you need out of the $q context, ie. `import { Dialog } from 'quasar'`, `setup() { Dialog.create(...) }`. I would prefer it this way (a bit of boilerplate) than using the context.root from setup param as that one is deprecated.
